lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -418,6 +453,56 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
     return success();
   }
 
+  LogicalResult handlePublicModuleRefPorts(FModuleOp module) {
+    // Decls must be at the top mlir module scope
+    auto declBuilder = ImplicitLocOpBuilder::atBlockBegin(
+        module.getLoc(), getOperation().getBody());
+    auto builder = ImplicitLocOpBuilder::atBlockBegin(module.getLoc(),
+                                                      circuitOp.getBodyBlock());
+
+    for (size_t portIndex = 0, numPorts = module.getNumPorts();
+         portIndex != numPorts; ++portIndex) {
+      auto refType = module.getPortType(portIndex).dyn_cast<RefType>();
+      if (!refType || isZeroWidth(refType.getType()) ||
+          module.getPortDirection(portIndex) != Direction::Out)
+        continue;
+      auto portValue =
+          module.getArgument(portIndex).cast<mlir::TypedValue<RefType>>();
+
+      Attribute ref;
+      SmallString<128> stringLeaf;
+      if (failed(resolveReferencePath(portValue, builder, ref, stringLeaf)))
+        return failure();
```

Comment:
This will error if the path ends above the current module, yes?

Reply:
What do you mean by that exactly?

Reply:
I want to make sure this won't try to generate a path if there is an output reference fed by an input reference that starts "above" the current module.

However, the parser will categorically reject input probes. There shouldn't be a problem here.

A circuit that could lead to weirdness is something like the following. It's the classic "there is no value that you can substitute for this path because it is more than one path" type thing:

```
circuit Foo :
  module Bar :
    input a: Probe<UInt<1>>
    output b: Probe<UInt<1>>

    define b = a

  module Foo :

    wire x: UInt<1>
    wire y: UInt<1>

    inst bar of Bar
    define bar.a = probe(x)

    inst bar2 of Bar
    define bar2.a = probe(y)
```